tests/test_user_prompt.py

```diff
@@ -0,0 +1,327 @@
+                                                   return_value=user_input)
+    patch_for_convert_path_string_to_list = mocker.patch("RUFAS.user_prompt.convert_path_string_to_list",
+                                                         return_value=[user_input])
+
+    # Act
+    actual = user_prompt()
+
+    # Assert
+    patch_for_prompt_user_for_input.assert_called_once()
+    patch_for_convert_path_string_to_list.assert_called_once_with(path=user_input)
+    assert actual == [user_input]
+
+
+def test_get_json_list_from_dir(mocker: MockerFixture,
+                                capsys: pytest.CaptureFixture):
+    """check that get_json_list_from_dir() properly detects all json files in a directory"""
+    verbose = True
+
+    # Case 1: Input path is not a directory
+    # Arrange
+    dummy_path = Path('dummy_path')
+
+    # Act
+    # check an error is raised
+    with pytest.raises(IsADirectoryError) as e:
+        get_json_list_from_dir(dummy_path, verbose)
+        assert "specified path is not a directory" in str(e.value)
+
+    # ------------------------------
+
+    # Case 2: Input path is a directory, but contains no json files
+    # Arrange
+    # Print current directory
+    dummy_dir = Path('dummy_dir')
+    dummy_dir.mkdir()
+    empty_json_paths = []
+    patch_for_glob = mocker.patch("RUFAS.user_prompt.Path.glob", return_value=empty_json_paths)
+
+    # Act
+    with pytest.raises(FileExistsError) as e:
+        get_json_list_from_dir(dummy_dir, verbose)
+        assert "Directory contains no json files" in str(e.value)
+
+    # Assert
+    patch_for_glob.assert_called_once_with('*.json')
+
+    # Cleanup
+    dummy_dir.rmdir()
+
+    # ------------------------------
+
+    # Case 3: Input path is a directory, and contains json files
+    # Arrange
+    dummy_dir = Path('dummy_dir')
+    dummy_dir.mkdir()
+    expected_json_paths = ['dummy.json', 'dummy2.json']
+    patch_for_glob = mocker.patch("RUFAS.user_prompt.Path.glob", return_value=expected_json_paths)
+
+    # Act
+    actual_json_paths = get_json_list_from_dir(dummy_dir, verbose)
+
+    # Assert
+    patch_for_glob.assert_called_once_with('*.json')
+    captured = capsys.readouterr()
+    assert f'{len(expected_json_paths)} json files detected...' in captured.out
+    assert actual_json_paths == expected_json_paths
+
+    # Cleanup
+    dummy_dir.rmdir()
+
+
+@pytest.mark.parametrize("path", [file_path])
+def test_convert_json_path_to_list(path):
+    """check that convert_json_path_to_list() properly returns a Path list from a json path string"""
+
+
+@pytest.mark.parametrize("user_input", ['Q', 'q', file_path, dir_path, 'dir', 'error'])
+def test_prompt_user_for_input(mocker: MockerFixture, capsys: pytest.CaptureFixture, user_input: str) -> None:
+    """check that prompt_user_for_input correctly returns the users input and that other options work as expected"""
+
+    # Arrange
+    def mock_user_input_error_side_effect(*args, **kwargs):
```

Comment:
@ltp23, are these arguments needed?

```suggestion
    def mock_user_input_error_side_effect():
```

Reply:
is this comment pending or resolved?

Reply:
I believe those two arguments are required in this case and they are used internally by `pytest`.